design/sketch.webidl

```diff
@@ -428,14 +428,46 @@ interface WebGPURenderPipeline {
 // COMMAND RECORDING (Command buffer and all relevant structures)
 // ****************************************************************************
 
-enum WebGPULoadOp {
+/// Common interface for render and compute pass encoders.
+interface WebGPUCommandEncoder {
+    WebGPUCommandBuffer end_pass();
+    // Allowed in both compute and render passes
+    void setPushConstants(WebGPUShaderStageFlags stage,
+                          u32 offset,
+                          u32 count,
+                          ArrayBuffer data);
+    void setBindGroup(u32 index, WebGPUBindGroup bindGroup);
+    void setPipeline((WebGPUComputePipeline or WebGPURenderPipeline) pipeline);
+};
+
+interface WebGPURenderPassEncoder: WebGPUCommandEncoder {
+    void setBlendColor(float r, float g, float b, float a);
+    void setIndexBuffer(WebGPUBuffer buffer, u32 offset);
+    void setVertexBuffers(u32 startSlot, sequence<WebGPUBuffer> buffers, sequence<u32> offsets);
+
+    void draw(u32 vertexCount, u32 instanceCount, u32 firstVertex, u32 firstInstance);
+    void drawIndexed(u32 indexCount, u32 instanceCount, u32 firstIndex, u32 firstInstance, u32 firstVertex);
+
+    // TODO add missing commands
+};
+
+interface WebGPUComputePassEncoder: WebGPUCommandEncoder {
+    void dispatch(u32 x, u32 y, u32 z);
```

Comment:
I don't think `u32` is a valid WebIDL type. https://heycam.github.io/webidl/

Reply:
There's this line at the top of the file.

> typedef unsigned long u32;

We need to think about the integer types at some point.